### PR TITLE
Improve command object documentation when you want to bind a list of objects. 

### PR DESCRIPTION
**Pull request by rpalcolea**
Improve command object documentation when you want to bind a list of objects. 

Issue https://github.com/grails/grails-doc/issues/487

Reported by @olavgg

### CODE REVIEW

File: src/en/guide/theWebLayer/controllers/commandObjects.gdoc [hunk 1]
```diff
@@ -205,3 +205,63 @@ class DemoController {
...
+class DemoController {
+
+    def createAuthor(AuthorCommand command) {
+        // ...
+
+    }
+
+    class AuthorCommand {
+        String fullName
+        List<BookCommand> books = []
```
Comment (jeffbrown) [inline src/en/guide/theWebLayer/controllers/commandObjects.gdoc]: FYI... `books` doesn't really have to be initialized here.  You could use `List<BookCommand> books` and the binder will initialize it for you.  Do you think that is worth noting?

File: src/en/guide/theWebLayer/controllers/commandObjects.gdoc [hunk 2]
```diff
@@ -205,3 +205,63 @@ class DemoController {
...
+    <g:fieldValue name="fullName" value=""/>
+    <ul>
+        <li>
+            <g:fieldValue name="books[0].title" value=""/>
+            <g:fieldValue name=""books[0].isbn" value=""/>
+        </li>
+
+        <li>
+            <g:fieldValue name="books[1].title" value=""/>
+            <g:fieldValue name=""books[1].isbn" value=""/>
```
Comment (jeffbrown) [inline src/en/guide/theWebLayer/controllers/commandObjects.gdoc]: There is a typo there with an extra double quotes on the `name` attribute.

File: src/en/guide/theWebLayer/controllers/commandObjects.gdoc [hunk 3]
```diff
@@ -205,3 +205,63 @@ class DemoController {
...
+
+In order to make this work from the UI layer, you can do the following in your GSP:
+
+{code:xml}
+<g:form name="submit-author-books" controller="demo" action="createAuthor">
+    <g:fieldValue name="fullName" value=""/>
+    <ul>
+        <li>
+            <g:fieldValue name="books[0].title" value=""/>
+            <g:fieldValue name=""books[0].isbn" value=""/>
```
Comment (jeffbrown) [inline src/en/guide/theWebLayer/controllers/commandObjects.gdoc]: There is a typo there with an extra double quotes on the name attribute.

**Comment by jeffbrown** [comment]
See my comments.  If you make any additional commits and push them to your `issue-487/commandObjects-lists` branch, those changes will automatically show up in this PR.

**Comment by rpalcolea** [comment]
Hi @jeffbrown thanks for the feedback and sorry for the typos.

Just fixed them

File: src/en/guide/theWebLayer/controllers/commandObjects.gdoc (hunk 1 above)
Comment (rpalcolea) [inline src/en/guide/theWebLayer/controllers/commandObjects.gdoc]: You are correct, my bad.

**Comment by jeffbrown** [comment]
I merged the PR and then put the changes on both the `3.0.x` and `3.1.x` branches.

**Pull request merged**